python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -399,12 +409,116 @@ def get_known_used_data(
             }
         return {}
 
+    def _calculate_partition_data_time(
```

Comment:
the name makes me think this is calculating the data time for a single partition - maybe `_calculate_time_partitioned_asset_data_time`?